### ISSUE TITLE
Broadcast based server locating

### ISSUE THREAD

**Issue by freyamade** [ExceptionalVoid/Arena#6]
We should implement a way for the lobby.py to find all open servers on the LAN, and whether or not they have passwords (depends on #9)

**Comment by crnbrdrck** [comment]
I will handle both server enhancements

**Comment by freyamade** [comment]
The form should stay, but only as a last resort

Or maybe private servers?

**Comment by freyamade** [comment]
Broadcast address: 255.255.255.255
Broadcast port: 44445

Have a separate thread to handle broadcast requests. Just one thread that is stored, since broadcast requests don't need to be handled suqer quickly

When a server receives a broadcast request on port 44445, it should reply to the client with the following info;
    Names of players in game
    Whether or not it is password protected (after #9)
    Have a join button

**Comment by freyamade** [comment]
Okay, we can't broadcast without UDP, and since UDP is blocked, we might have to postpone this issue

**Comment by freyamade** [comment]
I am an idiot. Python normally blocks posting to the broadcast IP. If you want to do it you have to first specify that you will be broadcasting;

s.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)

**Comment by freyamade** [comment]
I will fix this before doing the GUI

**Comment by freyamade** [comment]
How do we receive the data from all of the different servers?
Do we wait a certain number of seconds?
